backup/backup.py

```diff
@@ -289,57 +289,31 @@ def check_first_write(plugin, data_version):
 @plugin.hook('db_write')
 def on_db_write(writes, data_version, plugin, **kwargs):
     change = Change(data_version, None, writes)
-    if not hasattr(plugin, 'backend'):
-        plugin.early_writes.append(change)
-        return {"result": "continue"}
-    else:
-        return apply_write(plugin, change)
-
-
-def apply_write(plugin, change):
     if not plugin.initialized:
         assert(check_first_write(plugin, change.version))
         plugin.initialized = True
 
     if plugin.backend.add_change(change):
         return {"result": "continue"}
+    else:
+        kill("Could not append DB change to the backup. Need to shutdown!")
 
 
 @plugin.init()
-def on_init(options: Mapping[str, str], plugin: Plugin, **kwargs):
-    # Reach into the DB and
-    configs = plugin.rpc.listconfigs()
-    plugin.db_path = configs['wallet']
-    destination = options['backup-destination']
-
-    # Ensure that we don't inadventently switch the destination
-    if not os.path.exists("backup.lock"):
-        print("Files in the current directory {}".format(", ".join(os.listdir("."))))
-        kill("Could not find backup.lock in the lightning-dir, have you initialized using the backup-cli utility?")
-
-    d = json.load(open("backup.lock", 'r'))
-    if destination is None or destination == 'null':
-        destination = d['backend_url']
-    elif destination != d['backend_url']:
-        kill(
-            "The destination specified as option does not match the one "
-            "specified in backup.lock. Please check your settings"
+def on_init(options, **kwargs):
+    dest = options.get('backup-destination', 'null')
```

Comment:
We are missing this here, otherwise the sqlite3 check crashes :

```
    configs = plugin.rpc.listconfigs()
    plugin.db_path = configs['wallet']
```

Reply:
Ouch, right. Added the `listconfigs` call again, but not storing it in the `plugin` since we never use it again.